.github/workflows/rust.yml

```diff
@@ -49,7 +50,13 @@ jobs:
   cargo-clippy:
     strategy:
       matrix:
-        os: ${{ fromJson(github.repository_owner == 'autonomys' && '[["self-hosted", "ubuntu-22.04-x86-64"], ["self-hosted", "macos-14-arm64"], ["self-hosted", "windows-server-2022-x86-64"]]' || '["ubuntu-22.04", "macos-14", "windows-2022"]') }}
+        os: ${{ fromJson(github.repository_owner == 'autonomys' &&
+          '[
+            "runs-on=${{ github.run_id }}/runner=self-hosted-ubuntu-22.04-x86-64",
+            ["self-hosted", "windows-server-2022-x86-64"],
+            ["self-hosted", "macos-14-arm64"]
+          ]' ||
+          '["ubuntu-22.04", "macos-14", "windows-2022"]') }}
```

Comment:
Nit: the order here (and below) is Ubuntu, windows, Mac || Ubuntu, Mac, windows.

If the matrix order significantly impacts launch order, I’d prefer windows, Mac, Ubuntu, because that’s roughly longest to shortest run time. Otherwise I don’t think it matters, but consistency is nice.

Reply:
Different platforms run independently in parallel, so in that sense order doesn't matter. But consistency would be nice.

I guess some jobs failed because the configuration is not in `main` yet (documented on the website of the orchestration service used here).

Reply:
👍🏽 done